executor/executor.go

```diff
@@ -1567,7 +1594,14 @@ func (e *UnionExec) Close() error {
 		for range e.childIDChan {
 		}
 	}
-	return e.baseExecutor.Close()
+	// We do not need to acquire the e.mu.Lock since all the resultPuller can be
+	// promised to exit when reaching here (e.childIDChan been closed).
+	for i := 0; i <= e.mu.maxOpenedChildID; i++ {
```

Comment:
We should close all opened executors, and return firstErr. see `func (e *baseExecutor) Close()`